nvflare/client/api.py

```diff
@@ -34,6 +40,23 @@ class ClientAPIType(Enum):
 data_bus = DataBus()
 
 
+def death_watch():
+    """
+    Python's main thread doesn't die if there are running thread pools.
+    This function kills the process when the main thread is in the shutdown process
+    """
+    try:
+        while True:
+            if threading._SHUTTING_DOWN:
```

Comment:
I'm not sure this is a good practice. I googled about the use of SHUTTING_DOWN, and people seem to suggest against using it.

Maybe we should just make sure that all threads set daemon to True.